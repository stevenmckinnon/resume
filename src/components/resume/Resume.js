import React from 'react';
import skills from '../../cv-data/Skills';
import jobs from '../../cv-data/Jobs';
import education from '../../cv-data/Education';

class Resume extends React.Component {
    render() {
        return (
            <section id="resume">
                <div className="row work">
                    <div className="three columns header-col">
                        <h1><span>Work</span></h1>
                    </div>
                    <div className="nine columns main-col">
                        {jobs.map((job, index) => {
                            return (
                                <div key={index} className="row item">
                                    <div className="twelve columns">
                                        <h3>{job.company}</h3>
                                        <p className="info">{job.title} <span>•</span> <em
                                            className="date">{job.years}</em>
                                        </p>
                                        <p>{job.description}</p>
                                    </div>
                                </div>
                            )
                        })
                        }
                    </div>
                </div>

                <div className="row education">
                    <div className="three columns header-col">
                        <h1><span>Education</span></h1>
                    </div>
                    <div className="nine columns main-col">
                        {education.map((school, index) => {
                            return (
                                <div key={index} className="row item">
                                    <div className="twelve columns">
                                        <h3>{school.schoolName}</h3>
                                        <p className="info">{school.course}<span> • </span><em
                                            className="date">{school.years}</em></p>
                                    </div>
                                </div>
                            )
                        })
                        }
                    </div>
                </div>

                <div className="row skill">
                    <div className="three columns header-col">
                        <h1><span>Skills</span></h1>
                    </div>
                    <div className="nine columns main-col">
                        <div className="skills">
                            {skills.map((skill, index) => {
                                return (<div key={index} className="skill-tags">{skill.name}</div>)
                            })}
                        </div>
                    </div>
                </div>
            </section>
        );
    }
}

export default Resume;