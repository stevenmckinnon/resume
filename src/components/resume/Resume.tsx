import * as React from 'react';
import { format } from 'date-fns'
import { Job, Education } from '../../interfaces';
import { Element } from 'react-scroll';

interface Props {
    work: Job[];
    education: Education[];
    skills: string[];
}

const formatDates = (fromDate: string, toDate: string): string => {
    const from = format(new Date(fromDate.substring(0, 10)), 'MMMM yyyy');
    const to = toDate ? format(new Date(toDate.substring(0, 10)), 'MMMM yyyy') : 'Present';
    return `${from} - ${to}`;
}

const Resume: React.FC<Props> = ({ work, education, skills }) => (
    <Element name="resume">
        <section id="resume">
            <div className="row work">
                <div className="three columns header-col">
                    <h1><span>Work</span></h1>
                </div>
                <div className="nine columns main-col">
                    {
                        work.map((job, index) => {
                            console.log(job);
                            return (
                                <div key={index} className="row item">
                                    <div className="twelve columns">
                                        <h3>{job.company}</h3>
                                        <p className="info">{job.jobTitle} <span>•</span> <em
                                            className="date">{formatDates(job.fromDate, job.toDate)}</em>
                                        </p>
                                        <p>{job.description}</p>
                                    </div>
                                </div>
                            )
                        })
                    }
                </div>
            </div>

            <div className="row education">
                <div className="three columns header-col">
                    <h1><span>Education</span></h1>
                </div>
                <div className="nine columns main-col">
                    {education.map((school, index) => {
                        return (
                            <div key={index} className="row item">
                                <div className="twelve columns">
                                    <h3>{school.name}</h3>
                                    <p className="info">{school.course}<span> • </span><em
                                        className="date">{formatDates(school.fromDate, school.toDate)}</em></p>
                                </div>
                            </div>
                        )
                    })
                    }
                </div>
            </div>

            <div className="row skill">
                <div className="three columns header-col">
                    <h1><span>Skills</span></h1>
                </div>
                <div className="nine columns main-col">
                    <div className="skills">
                        {skills.map((skill, index) =>
                            <div key={index} className="skill-tags">{skill}</div>
                        )}
                    </div>
                </div>
            </div>
        </section>
    </Element>
);

export default Resume;