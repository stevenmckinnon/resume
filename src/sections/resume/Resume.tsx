import * as React from "react"
import { format } from "date-fns"
import { Job, Education } from "../../interfaces"
import { Element } from "react-scroll"

interface Props {
  work: Job[]
  education: Education[]
  skills: string[]
}

const formatDates = (fromDate: string, toDate: string): string => {
  const from = format(new Date(fromDate.substring(0, 10)), "MMMM yyyy")
  const to = toDate
    ? format(new Date(toDate.substring(0, 10)), "MMMM yyyy")
    : "Present"
  return `${from} - ${to}`
}

const Resume: React.FC<Props> = ({ work, education, skills }) => (
  <Element name="resume" className="bg-slate-50 pt-8 pb-8">
    <section id="resume" className="flex flex-col items-center">
      <div className="flex flex-col md:grid grid-cols-3 p-7 md:p-10 gap-6 max-w-4xl">
        <div className="col-span-1 flex justify-center">
          <h1 className="h-min text-2xl text-dark pb-1 border-b-4 border-teal-300 uppercase font-heading">
            Work
          </h1>
        </div>
        <div className="col-span-2">
          {work.map((job) => (
            <div key={job.fromDate} className="mb-8 flex flex-col gap-1">
              <h3 className="text-2xl text-primary font-bold font-body">
                {job.company}
              </h3>
              <h4 className="text-primary font-semibold font-body text-gray-500">
                {job.jobTitle} <span>•</span>{" "}
                <em className="date">{formatDates(job.fromDate, job.toDate)}</em>
              </h4>
              <p className="text-primary  font-body">{job.description}</p>
            </div>
          ))}
        </div>
      </div>

      <div className="flex flex-col md:grid grid-cols-3 p-7 md:p-10 gap-6 max-w-4xl">
        <div className="col-span-1 flex justify-center">
          <h1 className="h-min text-2xl text-dark pb-1 border-b-4 border-teal-300 uppercase font-heading">
            Education
          </h1>
        </div>
        <div className="col-span-2">
          {education.map((school) => (
            <div key={school.name} className="mb-8 flex flex-col gap-1">
              <div className="twelve columns">
                <h3 className="text-2xl text-primary font-bold font-body">
                  {school.name}
                </h3>
                <h4 className="text-primary font-semibold font-body text-gray-500">
                  {school.course}
                  <span> • </span>
                  <em className="date">
                    {formatDates(school.fromDate, school.toDate)}
                  </em>
                </h4>
              </div>
            </div>
          ))}
        </div>
      </div>

      <div className="flex flex-col md:grid grid-cols-3 p-7 md:p-10 pb-16 gap-6 max-w-4xl">
        <div className="col-span-1 flex justify-center">
          <h1 className="h-min text-2xl text-dark pb-1 border-b-4 border-teal-300 uppercase font-heading">
            Skills
          </h1>
        </div>
        <div className="col-span-2">
          <div className="flex flex-wrap justify-center items-center gap-3">
            {skills.map((skill, index) => (
              <div
                key={index}
                className="bg-teal-300 text-dark rounded-3xl p-4 font-body min-w-[100px]"
              >
                {skill}
              </div>
            ))}
          </div>
        </div>
      </div>
    </section>
  </Element>
)

export default Resume
