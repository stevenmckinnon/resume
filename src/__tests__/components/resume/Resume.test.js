import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Resume from '../../../components/resume/Resume';

Enzyme.configure({ adapter: new Adapter() });

describe('Tests the Resume component', () => {
    it('renders the Resume component', () => {
        const work = [
            {
                company: 'Company Name',
                description: 'Description of role',
                fromDate: '2018-06-20T00:00:00.000Z',
                toDate: null,
                jobTitle: 'Job title 1'
            },
            {
                company: 'Company Name 2',
                description: 'Description of role',
                fromDate: '2018-06-20T00:00:00.000Z',
                toDate: null,
                jobTitle: 'Job title 2'
            },
        ];
        const education = [
            {
              name: 'Uni name',
              course: 'Course name (BSc Hons)',
              fromDate: '2010-09-13T00:00:00.000Z',
              toDate: '2014-05-21T00:00:00.000Z'
            }
          ];
        const skills = [
            'JavaScript',
            'HTML',
            'CSS',
            'Partying'
          ];
        const wrapper = shallow(<Resume work={work} education={education} skills={skills} />);
        expect(wrapper).toMatchSnapshot();
    });
});
