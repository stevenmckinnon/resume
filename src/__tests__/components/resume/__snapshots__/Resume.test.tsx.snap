// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Resume component renders the Resume component 1`] = `ShallowWrapper {}`;
